### DISCUSSION TITLE
Daily Energy not reseting

### DISCUSSION THREAD

**Discussion by swetops** [klaasnicolaas/home-assistant-glow#137] [Q&A] [answered]
Hi.
Got everything up and running. But the Daily Energy isn't resetting daily.
Can't find in code how this is done,  only at bottom there's  the time component that seems to handle this.
The esp is on an isolated network, with no internet access. But that should not mess things up. Have tested to give internet access, still not resetting.

**Comment by ghitchen** [comment]
I have the same issue. I did try to work out why but couldn't see anything and then forgot about it. It would be good to resolve it though.

**Comment by klaasnicolaas** [comment]
Daily reset still works for me, which version of ESPHome are you running? And has it worked before?

**Reply by swetops** [reply]
Im on version 2022.1.3. I've also added an uptime sensor to the code. Otherwise no modification. Don't think it ever worked.
I also tried to add my internal ntp server which is reachable for the device. Still not working.
As I understand the time component handle the midnight reset and that's all I need. Can't figure why this won't work...

**Comment by robinelfving** [comment]
It depends on NTP for the time. Try to allow NTP (UDP123) to Internet or configure a local NTP server.

**Reply by swetops** [reply]
Thx, I will try open up the ntp to see if it helps.

**Reply by swetops** [reply]
Added a local ntp server without sucess....

**Comment by ghitchen** [comment]
The NTP access suggestion above made me think and now I have solved my issue with the daily energy not resetting. SNTP was looking for time servers by name. All my ESP devices have static IPs. I added DNS entries to the WiFi section and now it works.

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  manual_ip:
    # Set this to the IP of the ESP
    static_ip: 192.168.1.142
    # Set this to the IP address of the router. Often ends with .1
    gateway: 192.168.1.1
    # The subnet of the network. 255.255.255.0 works for most home networks.
    subnet: 255.255.255.0
    dns1: 8.8.8.8
    dns2: 8.8.4.4

**Reply by swetops** [reply]
Nice. But i guess this require the device to have internet access? Mine, like yours, also have static ip. But I want my device blocked from internet. I'll try open  up for internet and add dns to see if it works. But I doubt. Fond this on esp home page:

Note
If your are using [Manual IPs](https://esphome.io/components/wifi.html#wifi-manual-ip) make sure to configure a DNS Server (dns1, dns2) or use only IP addresses for the NTP servers.

So I guess it should be enough to add local nameserver by ip address. But dosen't work for me...

**Comment by swetops** [comment, marked as answer]
Anyone tried the home assistant time?  According to the esp website, this is the prefered way.
# Example configuration entry
time:
  - platform: homeassistant
    id: homeassistant_time

But I need to wait until midnight to know if it works

**Reply by ghitchen** [reply]
I've removed my DNS entries and changed time: to use this method. We will know the answer tomorrow. :-)

**Reply by swetops** [reply]
Yes. I can confirm that this work. Now my energy resets at midnight.

**Reply by ghitchen** [reply]
Yes, same here. Looks like this is a better way to get time as no Internet access is required on the ESP.
